graphenex/core/utils/helpers.py

```diff
@@ -177,3 +182,27 @@ def is_valid_address(address):
         return True
     except socket.gaierror:
         return False
+
+
+def create_env_file(cli_secret_key):
+    """Generate .env file with secret key"""
+
+    env_file_exists = env_file.is_file()
+    if not env_file_exists:
+        env_file.touch()
+        logger.info("`.env` config file created")
+
+    load_dotenv(env_file)
+    flask_secret_key = os.getenv("FLASK_SECRET_KEY")
+
+    if cli_secret_key or not flask_secret_key:
+        set_key(env_file, "FLASK_SECRET_KEY", cli_secret_key or os.urandom(32).hex())
```

Comment:
This approach is new to me - I've never seen any other project creating an env file with secret before. Is there any example apps that use this approach? What are the security implications?

Reply:
Honestly, I don't even know what the best approach would be here. In this case the `.env` file is useless for the user if he only uses CLI and not the Web UI. [The last time we talked about it](https://github.com/grapheneX/grapheneX/pull/198#discussion_r1614830289) we found the approach of checking for the `.env` file and logging error if it doesn't exist / secret key isn't set useless, because the user would have to go to docs and repeat the same process.

I found this approach the most "painless" for normal user as the process of creating the `.env` file and setting `FLASK_SECRET_KEY` defaultly with very secure value produced from `os.urandom` is all automated, and honestly the flask secret key is rarely even changed once it's set.

What would your approach for this be?

Reply:
In a way I'd also say that this isn't much different from what was set by default:

https://github.com/grapheneX/grapheneX/blob/b6025aaa8611319fe88941cefb4e8538e3d18063/graphenex/core/web/__init__.py#L16

Now the key is just set inside an `.env` file rather than hard-coded in the source code + Yeah,  I've also never seen setting a secret like that ever before, but for this purpose I'd say it isn't that bad.

Reply:
Maybe we can at least prompt user about this instead of directly creating it? Or at least add a log message about it? I would be personally annoyed if any tool I'm trying out creates a sensitive file as default without letting me know.

Reply:
Yes, or we could just straight up refuse to launch gX, If the user doesn't set the secret key in the `.env` file, but like I said It is useless for CLI users who do not use web.

Reply:
Yeah, I think the way to go is adding a log line such as "creating .env" and we can just yolo it.

Reply:
> Yeah, I think the way to go is adding a log line such as "creating .env" and we can just yolo it.

Then I guess this PR is ready to get merged

![Screenshot from 2024-06-22 21-46-01](https://github.com/grapheneX/grapheneX/assets/138708600/22a8b08f-4cd3-4fca-81d1-dd21ce3ab23f)